api/v1alpha1/componentversion_types.go

```diff
@@ -102,6 +102,23 @@ type ComponentVersionStatus struct {
 
 	ReconciledVersion string `json:"reconciledVersion,omitempty"`
 	Verified          bool   `json:"verified,omitempty"`
+
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the last reconciled generation.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+
+// GetConditions returns the conditions of the ComponentVersion.
+func (in *ComponentVersion) GetConditions() []metav1.Condition {
+	return in.Status.Conditions
+}
+
+// SetConditions sets the conditions of the ComponentVersion.
+func (in *ComponentVersion) SetConditions(conditions []metav1.Condition) {
+	in.Status.Conditions = conditions
 }
```

Comment:
Can we add a kubebuilder annotation here to show the Ready condition as a status field via kubectl?

Something like this:
```golang
// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.conditions[?(@.type==\"Ready\")].status",description=""
// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.conditions[?(@.type==\"Ready\")].message",description=""